presto-orc/src/main/java/com/facebook/presto/orc/OrcRecordReader.java

```diff
@@ -393,12 +398,70 @@ private void advanceToNextStripe()
         for (int columnId = 0; columnId < rowType.getFieldCount(); columnId++) {
             if (includedColumns.containsKey(columnId)) {
                 StreamDescriptor streamDescriptor = streamDescriptors.get(columnId);
+
+                Type type = includedColumns.get(columnId);
+                if (!validateOrcType(streamDescriptor, type.getTypeSignature())) {
+                    throw new OrcCorruptionException("ORC type from file footer (%s) does not match expected type of table (%s)", types, includedColumns);
+                }
+
                 streamReaders[columnId] = StreamReaders.createStreamReader(streamDescriptor, hiveStorageTimeZone);
             }
         }
         return streamReaders;
     }
 
+    private static boolean validateOrcType(StreamDescriptor streamDescriptor, TypeSignature type)
```

Comment:
Name this `validOrcType` (short for `isValidOrcType`) because it is returning a boolean. A method named "validate" indicates that it does validation and will throw an exception on failure.